active_directory_ldap/komand_active_directory_ldap/util/utils.py

```diff
@@ -144,3 +148,66 @@ def escape_brackets_for_query(query: str, pairs: dict) -> str:
                 query = query[:value] + "\\29" + query[value + 1 :]
                 query = query[:key] + "\\28" + query[key + 1 :]
         return query
+
+    @staticmethod
+    def escape_user_dn(user_dn: str) -> str:
+        pairs = ADUtils.find_parentheses_pairs(user_dn)
+        if pairs:
+            # replace ( and ) when they are part of a name rather than a search parameter
+            user_dn = ADUtils.escape_brackets_for_query(user_dn, pairs)
+
+        return user_dn
+
+    @staticmethod
+    def check_user_dn_is_valid(conn, user_dn: str, search_base: str) -> bool:
+        try:
+            conn.search(
+                search_base=search_base,
+                search_filter=f'(distinguishedName={ADUtils.escape_user_dn(user_dn)})',
+                attributes=['userAccountControl']
+            )
+        except LDAPInvalidDnError as e:
+            raise PluginException(
+                cause='The server not found group.',
```

Comment:
Could we be a bit clearer with this message?  Even looking at the code I'm unsure what the problem could be.  We're in a check_user_dn_is_valid function and throwing an error talking about groups.

Reply:
I changed this message. We can see this in: `tests/enable_user_bad.json` :)